File: src/Scripting/Ruby/patterns/mvc.rb
# frozen_string_literal: true

module RubyPatterns
  module MVC
    module_function

    def run
      model = { count: 0 }
      controller = -> { model[:count] += 1 }
      view = -> { "count=#{model[:count]}" }
      controller.call
      raise 'MVC failed' unless view.call == 'count=1'
      true
    end
  end
end

RubyPatterns::MVC.run if $PROGRAM_NAME == __FILE__

File: src/Scripting/Lua/patterns/mvc.lua
local model = { count = 0 }
local view = { render = function(m) return "count=" .. m.count end }
local controller = { increment = function(m) m.count = m.count + 1 end }
controller.increment(model)
assert(view.render(model) == "count=1")
return true
